components/ca-input-array.js

```diff
@@ -0,0 +1,631 @@
+            }
+
+            if (this.autofocus) this.setFocus(0);
+        }
+
+        /**
+         * @param {event} e event
+         * @returns {void}
+         */
+        removeClick(e) {
+
+            const ev = e || event;
+            const el = ev.target || ev.srcElement;
+            const index = parseInt(el.parentElement.getAttribute('data-index') || '0', 10);
+
+            this.removeItem(index);
+
+            cancelEvent(ev);
+        }
+
+        /**
+         * @param {event} e event
+         * @returns {void}
+         */
+        addClick(e) {
+
+            const ev = e || event;
+            const el = ev.target || ev.srcElement;
+            const index = parseInt(el.parentElement.getAttribute('data-index') || '0', 10);
+
+            this.addItem(index + 1);
+
+            cancelEvent(ev);
+        }
+
+        /**
+         * handles input on key press events - http://unixpapa.com/js/key.html
+         * @param {event} e event
+         * @returns {void}
+         */
+        keypressHandler(e) {
+
+            const ev = e || event;
+            const el = ev.target || ev.srcElement;
+            const key = ev.charCode || ev.keyCode || ev.which || '';
+            const char = String.fromCharCode(key);
+            const index = parseInt(el.parentElement.getAttribute('data-index') || '0', 10);
+
+            // respond to key events
+            switch (key) {
+
+                case 13: { // handle ENTER key press
+
+                    cancelEvent(ev);
+
+                    // if we have no max items OR the current count < maxItem && index == this.count (at the bottom)
+                    if ((this.maxItems === -1 || this.count < this.maxItems) && index === this.count - 1) {
+                        this.addItem(-1);
+                    } else {
+                        // move to next item, or remain at the bottom
+                        this.setFocus((this.count > index) ? index + 1 : index);
+                    }
+
+                } break;
+
+                case 8: { // handle BACKSPACE key press
+
+                    if (el.value === '') {
+
+                        if (this.count > this.minItems) {
+
+                            cancelEvent(ev);
+                            this.removeItem(index);
+                        } else {
+
+                            // move to previous item
+                            this.setFocus((index > 0) ? index - 1 : index);
+                        }
+                    }
+                } break;
+
+                default: {
+                    if (this.type === 'number' && !this.isNavKey(key) && !this.isNumberChar(char)) {
+                        cancelEvent(ev);
+                    }
+                }
+            }
+        }
+
+        /**
+         * determines if a navigation key has been passed as key
+         * @param {integer} key key
+         * @returns {boolean} is navigation key
+         */
+        isNavKey(key) {
```

Comment:
I have already exported this as a helper in https://github.com/cambridgeweblab/common-ui/pull/36 -- I think it would be wise to use this helper as it's used in 2 components